tests/b902_extended.py

```diff
@@ -2,10 +2,22 @@
 class Errors:
     # correctly registered as classmethod
     @validator
-    def foo1(self) -> None:
+    def foo_validator(self) -> None:
```

Comment:
I think we also need to test that non-default names work, including dotted attribute names, to check for compatibility problems with pep8-naming.